### DISCUSSION TITLE
Drawer

### DISCUSSION THREAD

**Discussion by alizedebray** [swisspost/design-system#2287] [Ideas]
A drawer is a pane that spans a container on one of its sides.
It can either be modal or non modal and might be resized by the user. 

**Use cases**
- Side navigation: navigation and main content are side by side 
![image](https://github.com/swisspost/design-system/assets/33580481/c84f4e75-ea2c-4f18-bf56-d1a90679d6fc)

- Edition pane: edition pane is next or over the main content, the main content is still reachable

**Variants**
- Close
- Reduce
- Scroll
- Slide over
 

**Resources**
- w3: https://www.w3.org/WAI/ARIA/apg/patterns/windowsplitter/
- Shoelace: https://shoelace.style/components/drawer & https://shoelace.style/components/split-panel

**Comment by gfellerph** [comment]
I see at least two components here, also because they will not necessarily share semantics or functionality.

Side navigation with split panel:
```html
<body>
  <post-split vertical>
    <post-split-panel>
      <aside>
        <post-tree>
          <post-tree-item href="/home">Home</post-tree-item>
        </post-tree>
      </aside>
    </post-split-panel>
    <post-split-panel>
      <main>
        lorem etc.
      </main>
    </post-split-panel>
  </post-split>
</body>
```

Side navigation with drawer:
```html
<body>
  <post-drawer modal placement="left">
    <aside>
      <post-tree>
        <post-tree-item href="/"">Home</post-tree-item>
      </post-tree>
    </aside>
  </post-drawer>

  <main>
    lorem etc.
  </main>
</body>
```
